### PR TITLE
Added meta map, added nf-test, minor code quality improvements

### LINKED ISSUE #5150: Add meta to iqtree
### Is your feature request related to a problem? Please describe

It is hard to perform batch processing with this module, as it does not have a meta input/output. Output files cannot be traced back to their origin.

### Describe the solution you'd like

Add a meta map.

### Describe alternatives you've considered

We could use regex on output files but no.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by itrujnara**
## PR checklist

Closes #5150 

- [x] Added meta in order to facilitate batch analysis. Added tests.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/iqtree/main.nf [hunk 1]
```diff
@@ -8,20 +8,21 @@ process IQTREE {
...
 
     input:
-    path alignment
+    tuple val(meta), path(alignment)
     val constant_sites
 
     output:
-    path "*.treefile",    emit: phylogeny
-    path "versions.yml" , emit: versions
+    tuple val(meta), path("*.treefile") , emit: phylogeny
```
Comment (lrauschning) [inline modules/nf-core/iqtree/main.nf:15]: As we talked about, I could see users wanting to use the bootstrap tree output. Might be worth to add another channel for this and document that the channel is empty unless `bootstrap` is passed in the args (or even to make it a value input channel, not sure what's better).

File: modules/nf-core/iqtree/meta.yml [hunk 2]
```diff
@@ -13,10 +13,19 @@ tools:
...
+        Groovy map containing sample information
+        e.g. [ id: 'test' ]
   - alignment:
       type: file
       description: A FASTA format multiple sequence alignment file
       pattern: "*.{fasta,fas,fa,mfa}"
+  - constant_sites:
+      type: string
+      description: Number of constant sites to add,
+        see iqtree documentation for details
```
[the remark this answers is not in the document]
Comment (itrujnara) [reply modules/nf-core/iqtree/meta.yml:28]: Added

File: modules/nf-core/iqtree/main.nf [hunk 3]
```diff
@@ -31,6 +32,8 @@ process IQTREE {
         -ntmax $task.cpus \\
         -mem $memory \\
```
[the remark this answers is not in the document]
Comment (itrujnara) [reply modules/nf-core/iqtree/main.nf:35]: It is, main.nf line 30

File: modules/nf-core/iqtree/main.nf [hunk 4]
```diff
@@ -31,6 +32,8 @@ process IQTREE {
...
 
+    mv ${alignment}.treefile ${prefix}.treefile
```
[the remark this answers is not in the document]
Comment (itrujnara) [reply modules/nf-core/iqtree/main.nf]: It actually has one, I have changed it to use the iqtree option instead

File: modules/nf-core/iqtree/main.nf (hunk 1 above)
Comment (itrujnara) [reply modules/nf-core/iqtree/main.nf:15]: Added bootstrap output

File: modules/nf-core/iqtree/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                    file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin.ref") ]
+                input[1] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.phylogeny }
```
[the remark this answers is not in the document]
Comment (itrujnara) [reply modules/nf-core/iqtree/tests/main.nf.test]: Added

**Comment by itrujnara** [comment]
CI note: there seems to be a random error in IQTREE, it is marked as resolved in the program's repo but seems to occur here as well. Interestingly enough, I have not had this issue locally.

**Comment by lrauschning** [comment]
eh sorry, 2.2.2.6

**Review by lrauschning — APPROVED** [review]

**Pull request merged**